### ISSUE TITLE
Contrastive Adversarial Training

### ISSUE THREAD

**Issue by CancanZhang** [RU-System-Software-and-Security/BppAttack#3]
Hi,

The paper said "The whole training framework follows the contrastive learning framework" as the paper "Supervised Contrastive Learning", but I cannot find the corresponding loss function in the bppattack.py file.

Could you please point out which part is "**Supervised Contrastive Learning**"? After I checked the bppattack.py file, I found only the cross-entropy loss of the input ([bad_input, negative_input, input]) and their corresponding label are used. 

Did you put the vanilla training file instead of the contrastive adversarial training file? If so, could you please upload a contrastive adversarial training file?

Let me know if I understand it wrong. Thanks!

**Comment by ZhentingWang** [comment]
Hi, 

Thanks for your interest in this work. 
You can find the code for the contrastive loss in the following repo:
https://github.com/HobbitLong/SupContrast
We will upload the contrastive training file later and let you know once it is done.

Best,
:)

**Comment by CancanZhang** [comment]
Thanks!
